catkin_virtualenv/cmake/catkin_generate_virtualenv.cmake

```diff
@@ -141,16 +141,29 @@ function(catkin_generate_virtualenv)
       install/${venv_dir}
   )
 
-  add_custom_target(venv_lock
-    COMMENT "Manually invoked target to generate the lock file on demand"
-    COMMAND ${CATKIN_ENV} rosrun catkin_virtualenv venv_lock ${CMAKE_BINARY_DIR}/${venv_dir}
-      --package-name ${PROJECT_NAME} --input-requirements ${ARG_INPUT_REQUIREMENTS}
-      --extra-pip-args ${processed_pip_args}
-    WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
-    DEPENDS
-      ${venv_devel_dir}
-      ${CMAKE_SOURCE_DIR}/${ARG_INPUT_REQUIREMENTS}
-  )
+  if(CMAKE_CURRENT_SOURCE_DIR STREQUAL CMAKE_SOURCE_DIR)
+    add_custom_target(venv_lock
+      COMMENT "Manually invoked target to generate the lock file on demand"
+      COMMAND ${CATKIN_ENV} rosrun catkin_virtualenv venv_lock ${CMAKE_BINARY_DIR}/${venv_dir}
+        --package-name ${PROJECT_NAME} --input-requirements ${ARG_INPUT_REQUIREMENTS}
+        --extra-pip-args ${processed_pip_args}
+      WORKING_DIRECTORY ${CMAKE_SOURCE_DIR}
+      DEPENDS
+        ${venv_devel_dir}
+        ${CMAKE_SOURCE_DIR}/${ARG_INPUT_REQUIREMENTS}
+    )
+  else()
+    add_custom_target(${PROJECT_NAME}_venv_lock
```

Comment:
I'm a little surprised if this is the only change required for non-isolated builds to work, but regardless, no need to put it behind an `if`, feel free to just change the `venv_lock` target name to `${PROJECT_NAME}_venv_lock` here and in the docs.

Reply:
> I can combine them and update the docs

I think I'd prefer this, since it would avoid any special cases that need extra testing.

Reply:
Will do.